### PR TITLE
mariadb_server: New attributes, flatten `storage_profile` block

### PR DESCRIPTION
**Pull request by manicminer**
- `create_mode`, `creation_source_server_id`, `public_network_access_enabled`

- Support replicas and point-in-time restores

- Use booleans for toggle properties, renaming
  * `auto_grow` -> `auto_grow_enabled`
  * `geo_redundant_backup` -> `geo_redundant_backup_enabled`
  * `ssl_enforcement` -> `ssl_enforcement_enabled`

- Remove redundant `storage_profile` block and move all properties
  within to top level

- Remove unpopulated attribute `administrator_login_password` from
  `mariadb_server` data source

Fixes: #4296, #6521
Related: #6459

### CODE REVIEW

File: azurerm/internal/services/mariadb/mariadb_virtual_network_rule_resource.go [hunk 1]
```diff
@@ -7,14 +7,15 @@ import (
 	"strings"
 	"time"
 
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mariadb/validate"
```
Comment (mbfrahry) [inline azurerm/internal/services/mariadb/mariadb_virtual_network_rule_resource.go:10]: We'll want to move this down into the other imports

File: azurerm/internal/services/mariadb/tests/mariadb_server_resource_test.go [hunk 2]
```diff
@@ -11,73 +12,109 @@ import (
...
+					resource.TestCheckResourceAttr(data.ResourceName, "version", version),
 				),
 			},
 			data.ImportStep("administrator_login_password"), // not returned as sensitive
 		},
 	})
 }
 
-func TestAccAzureRMMariaDbServer_requiresImport(t *testing.T) {
+func TestAccAzureRMMariaDbServer_basicTenTwoDeprecated(t *testing.T) {
```
Comment (mbfrahry) [inline azurerm/internal/services/mariadb/tests/mariadb_server_resource_test.go:36]: Let's add a note to remove this and any other deprecated test with 3.0

File: azurerm/internal/services/mariadb/mariadb_server_resource.go [hunk 3]
```diff
@@ -71,101 +154,119 @@ func resourceArmMariaDbServer() *schema.Resource {
...
-				ForceNew: true,
-				ValidateFunc: validation.StringInSlice([]string{
-					"10.2",
-					"10.3",
-				}, false),
+			"storage_mb": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"storage_profile", "storage_profile.0.storage_mb"},
```
[the remark this answers is not in the document]
Comment (manicminer) [reply azurerm/internal/services/mariadb/mariadb_server_resource.go:181]: Ah, makes sense, thanks!

**Comment by manicminer** [comment]
@mbfrahry thanks for the feedback! :)

**Pull request merged**